src/test/java/org/graylog/integrations/notifications/types/SlackEventNotificationTest.java

```diff
@@ -0,0 +1,167 @@
+package org.graylog.integrations.notifications.types;
+
+import com.floreysoft.jmte.Engine;
+import com.google.common.collect.ImmutableList;
+import org.graylog.events.notifications.*;
+import org.graylog.events.notifications.types.HTTPEventNotificationConfig;
+import org.graylog2.notifications.Notification;
+import org.graylog2.notifications.NotificationImpl;
+import org.graylog2.notifications.NotificationService;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.MessageSummary;
+import org.graylog2.plugin.system.NodeId;
+import org.graylog2.shared.bindings.providers.ObjectMapperProvider;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SlackEventNotificationTest extends SlackPluginTestFixture {
+
+    @Mock
+    NodeId mockNodeId;
+    private SlackEventNotification slackEventNotification;
+    private SlackEventNotificationConfig slackEventNotificationConfig;
+    private EventNotificationContext eventNotificationContext;
+
+    public SlackEventNotificationTest() throws IOException {
+    }
+
+    @Before
+    public void setUp() {
+
+        setSlackEventNotificationConfig();
+        eventNotificationContext = NotificationTestData.getDummyContext(getHttpNotification(), "ayirp").toBuilder().notificationConfig(slackEventNotificationConfig).build();
+
+
+        final ImmutableList<MessageSummary> messageSummaries = ImmutableList.of(
+                new MessageSummary("graylog_1", new Message("Test message 1", "source1", new DateTime(2020, 9, 6, 17, 0, DateTimeZone.UTC))),
+                new MessageSummary("graylog_2", new Message("Test message 2", "source2", new DateTime(2020, 9, 6, 17, 0, DateTimeZone.UTC)))
+        );
+        EventNotificationService notificationCallbackService = mock(EventNotificationService.class);
+        when(notificationCallbackService.getBacklogForEvent(eventNotificationContext)).thenReturn(messageSummaries);
+
+        NotificationService mockNotificationService = mock(NotificationService.class);
+        SlackClient slackClient = new SlackClient(getOkHttpClient());
```

Comment:
As mentioned above, you should be mocking the `SlackClient` for this test.

Reply:
i will give it a shot. probably, if it ends up having to remove the base class.